### PR TITLE
Set manual `tabindex`es on login page

### LINKED ISSUE #31686: Login form not password manager friendly
### Description

Since #31530 the login form can't be used with external password managers (or simply behaviour driven users) which type `{username}[TAB]{password}[ENTER]` because the `Forgot password?` label gets the focus after pressing the tab key.

### Screenshots

grafik

### PR DESCRIPTION
**Pull request by silverwind**
Fixes https://github.com/go-gitea/gitea/issues/31686.

A more elborate manual tabindex numbering could be done, but I think it's not really worth the extra effort and such stuff could easily break during refactors.

Includes another small tweak to un-stretch the`<a>` element so it's only as large as it needs to be and this change also made the margin unneeded.

### CODE REVIEW

File: templates/user/auth/signin_inner.tmpl
Comment (wxiaoguang) [inline templates/user/auth/signin_inner.tmpl:24]: Why a link should be inaccessible by the keyboard?

It breaks the keyboard-users and screen readers.

Comment (KN4CK3R) [reply templates/user/auth/signin_inner.tmpl:24]: Yes, I found the `tabindex` too but don't think it's a good solution. Other [logins](https://github.com/login) (Github, IMDb) use the same structure but the link does not get the focus even without a `tabindex`.

Comment (silverwind) [reply templates/user/auth/signin_inner.tmpl:24]: GitHub uses some odd `position: absolute` to get around the manual `tabindex`. Maybe we could do too, but we are having a number of constraints here, more than they do.

Comment (KN4CK3R) [reply templates/user/auth/signin_inner.tmpl:24]: The [IMDb](https://www.imdb.com/ap/signin?openid.pape.max_auth_age=0&openid.return_to=https%3A%2F%2Fwww.imdb.com%2Fregistration%2Fap-signin-handler%2Fimdb_us&openid.identity=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.assoc_handle=imdb_us&openid.mode=checkid_setup&siteState=eyJvcGVuaWQuYXNzb2NfaGFuZGxlIjoiaW1kYl91cyIsInJlZGlyZWN0VG8iOiJodHRwczovL3d3dy5pbWRiLmNvbS8_cmVmXz1sb2dpbiJ9&openid.claimed_id=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0%2Fidentifier_select&openid.ns=http%3A%2F%2Fspecs.openid.net%2Fauth%2F2.0&tag=imdbtag_reg-20) login doesn't have a `position: absolute`. Can't see what trick they use.

Comment (silverwind) [reply templates/user/auth/signin_inner.tmpl:24]: Interesting. By natural HTML order, the link should focus first there, but does not. Can't see anything immediately obvious that would influence the tabbing order either.

Comment (silverwind) [reply templates/user/auth/signin_inner.tmpl:24]: Actually I see the imdb page is just using manual `tabindex`.

1. Username
2. Password
3. Sign In

I guess in our case we have to insert "Remember me" in position 3.

Comment (silverwind) [reply templates/user/auth/signin_inner.tmpl:24]: Done that in https://github.com/go-gitea/gitea/pull/31689/commits/9d7e84233691b808e7eb8172891a868b9286e0c4, so for us it's 

1. Username
2. Password
3. Remember Me
4. Sign In

Comment (silverwind) [reply templates/user/auth/signin_inner.tmpl:24]: Tweaked once more:

1. Username
2. Password
3. Sign In
4. Forgot Password
5. Remember Me

I think it's good to optimize the common case of username,password,signin. We are lacking a keyboard focus style on buttons which is a accessibility concern, but we can fix that another time.

**Comment by delvh** [comment]
The only way I'll approve this PR as it is is if you document somewhere in that file exactly why we have this weird tab order.
Alternatively, we could also think about moving the `forgot password` label below the entry.

**Comment by silverwind** [comment]
Yes, I thought about adding a comment so this does not break during refactoring. I don't think it's weird, it's more like "optimized for usability" and the position for the forgot link is ideal and follows what other login forms do as well.

**Comment by delvh** [comment]
> weird tab order

I meant for someone who looks over the code and hasn't seen this PR.
For them, this tab order is indeed weird

**Review by techknowlogick — DISMISSED** [review]
not blocking the PR, just having this here so that it doesn't get merged prior to the merger going through the comments. feel free to dismiss this once discussion is complete

**Review by KN4CK3R — APPROVED** [review]
Does what I expect.

Video for others:

**Review by lunny — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Pull request merged**
